### ISSUE TITLE
Bignum: Reuse the new exponentiation algorithm for classic MPIs

### ISSUE THREAD

**Issue by yanesca** [ARMmbed/mbedtls#6600]
### Justification

Mbed TLS needs this because it will save code size, maintenance effort and remove risk.

### Suggested enhancement

Use `mbedtls_mpi_core_exp_mod()` to implement `mbedtls_mpi_exp_mod()`.

Once the PR is ready but before reviews start, measure the impact of the PR as described in the non-functional testing documentation in /kb/how-to/non-functional-testing.md in the [Mbed TLS Docs repository](https://github.com/Mbed-TLS/mbedtls-docs). 

If it is worse than expected,  raise a new issue for considering mitigations or alternative solutions.

The task is done when

- [ ] The measurements are posted in a comment to this issue
- [ ] Depending on the measurement outcome, either the PR is merged or a new issue is raised
